src/js/letters/actions/letters.js

```diff
@@ -21,17 +19,76 @@ export function getBenefitSummaryOptions() {
     apiRequest(
       '/v0/letters/beneficiary',
       null,
-      (response) => {
-        return dispatch({
-          type: 'GET_BENEFIT_SUMMARY_OPTIONS_SUCCESS',
-          data: response,
+      response => dispatch({
+        type: 'GET_BENEFIT_SUMMARY_OPTIONS_SUCCESS',
+        data: response,
+      }),
+      () => dispatch({ type: 'GET_BENEFIT_SUMMARY_OPTIONS_FAILURE' })
+    );
+  };
+}
+
+export function getLetterPdf(letterType, letterName, letterOptions) {
+  let settings;
+  if (letterType === 'benefit_summary') {
+    settings = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(letterOptions)
+    };
+  } else {
+    settings = {
+      method: 'POST'
+    };
+  }
+
+  // We handle IE10 separately but assume all other vets.gov-supported
+  // browsers have blob URL support.
+  // TODO: possibly want to explicitly check for blob URL support with something like
+  // const blobSupported = !!(/^blob:/.exec(downloadUrl));
+  const ie10 = !!window.navigator.msSaveOrOpenBlob;
+  const save = document.createElement('a');
+  let downloadWindow;
+  const downloadSupported = typeof save.download !== 'undefined';
+  if (!downloadSupported) {
+    // Instead of giving the file a readable name and downloading
+    // it directly, open it in a new window with an ugly hash URL
+    downloadWindow = window.open();
+  }
+  let downloadUrl;
+  return (dispatch) => {
+    apiRequest(
+      `/v0/letters/${letterType}`,
+      settings,
+      response => {
+        response.blob().then(blob => {
+          if (ie10) {
+            window.navigator.msSaveOrOpenBlob(blob, letterName);
+          } else {
+            window.URL = window.URL || window.webkitURL;
+            downloadUrl = window.URL.createObjectURL(blob);
+            if (downloadSupported) {
+              // Give the file a readable name if the download attribute is supported.
+              save.download = letterName;
+              save.href = downloadUrl;
+              save.target = '_blank';
+              document.body.appendChild(save);
+              save.click();
+              document.body.removeChild(save);
+            } else {
+              downloadWindow.location.href = downloadUrl;
+            }
+          }
         });
+        window.URL.revokeObjectURL(downloadUrl); // make sure this doesn't cause problems
+        dispatch({ type: 'GET_LETTER_PDF_SUCCESS' });
       },
-      () => dispatch({ type: 'GET_BENEFIT_SUMMARY_OPTIONS_FAILURE' })
+      () => dispatch({ type: 'GET_LETTER_PDF_FAILURE', data: letterType })
```

Comment:
Yes, great point. Probably a question for @rroueche and @phiden. I assume display some kind of error message to the user?